crates/ruff_linter/src/rules/flake8_errmsg/rules/string_in_exception.rs

```diff
@@ -269,28 +270,27 @@ pub(crate) fn string_in_exception(checker: &mut Checker, stmt: &Stmt, exc: &Expr
 fn generate_fix(
     stmt: &Stmt,
     exc_arg: &Expr,
-    indentation: &str,
+    stmt_indentation: &str,
     stylist: &Stylist,
-    generator: Generator,
+    locator: &Locator,
 ) -> Fix {
-    let assignment = Stmt::Assign(ast::StmtAssign {
-        targets: vec![Expr::Name(ast::ExprName {
-            id: "msg".into(),
-            ctx: ExprContext::Store,
-            range: TextRange::default(),
-        })],
-        value: Box::new(exc_arg.clone()),
-        range: TextRange::default(),
-    });
-
     Fix::unsafe_edits(
         Edit::insertion(
-            format!(
-                "{}{}{}",
-                generator.stmt(&assignment),
-                stylist.line_ending().as_str(),
-                indentation,
-            ),
+            if locator.contains_line_break(exc_arg.range()) {
+                format!(
+                    "msg = ({line_ending}{stmt_indentation}{indentation}{}{line_ending}{stmt_indentation}){line_ending}{stmt_indentation}",
+                    locator.slice(exc_arg.range()),
+                    line_ending = stylist.line_ending().as_str(),
+                    indentation = stylist.indentation().as_str(),
+                )
```

Comment:
Is this too much? 😅 

We can just keep a simple `format!("msg = ({}){}{}")` instead.